### PR TITLE
Print the state when the align instructions don't match up

### LINKED ISSUE #82729: Assertion failed '(igInLoop->igLoopBackEdge == nullptr) || (igInLoop->igLoopBackEdge == igLoopHeader)'
Libraries pgo, possibly recent regression in weekend runs (I had clean private runs on Friday, but this is PGO so may not always be testing the exact same codegen)

https://dev.azure.com/dnceng-public/public/_build/results?buildId=185788&view=ms.vss-test-web.build-test-results-tab

```
export DOTNET_TieredCompilation=1
export DOTNET_DbgEnableMiniDump=1
export DOTNET_EnableCrashReport=1
export DOTNET_DbgMiniDumpName=$HELIX_DUMP_FOLDER/coredump.%d.dmp
export DOTNET_ReadyToRun=0
export DOTNET_TC_QuickJitForLoops=1
export DOTNET_TieredPGO=1
export DOTNET_JitRandomGuardedDevirtualization=1
export DOTNET_JitRandomEdgeCounts=1
export DOTNET_JitRandomlyCollect64BitCounts=1

dotnet exec --runtimeconfig System.Runtime.Serialization.Xml.Tests.runtimeconfig.json --depsfile System.Runtime.Serialization.Xml.Tests.deps.json xunit.console.dll System.Runtime.Serialization.Xml.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing 
popd
===========================================================================================================
/root/helix/work/workitem/e /root/helix/work/workitem/e
  Discovering: System.Runtime.Serialization.Xml.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Serialization.Xml.Tests (found 278 of 279 test cases)
  Starting:    System.Runtime.Serialization.Xml.Tests (parallel test collections = on, max threads = 2)

Assert failure(PID 27 [0x0000001b], Thread: 33 [0x0021]): Assertion failed '(igInLoop->igLoopBackEdge == nullptr) || (igInLoop->igLoopBackEdge == igLoopHeader)' in 'System.Enum:TryParseByValueOrName[int](System.RuntimeType,System.ReadOnlySpan`1[ushort],bool,bool,byref):bool' during 'Generate code' (IL size 724; hash 0xef12aa33; Tier1)

    File: /__w/1/s/src/coreclr/jit/emit.cpp Line: 5713
    Image: /root/helix/work/correlation/dotnet
```

### PR DESCRIPTION
**Pull request by kunalspathak**
In https://github.com/dotnet/runtime/issues/82729, we hit a problem where we have `align` instruction for a loop that also contains an `align` instruction for a different loop. This state is not permitted. So, print the state of loops with relevant information.  ~Most likely, the fix for such cases would be in `placeLoopAlignInstructions()` but it is hard to determine the missing case without a repro.~ This was happening because during block compaction, we were failing to update the `bbNatLoopNum` of a block `A` into which the other block `B` is getting compacted into. Block `B` turned out to be backward jump source of a loop. Block `A` was placed lexically before the top-loop. In `placeLoopAlignInstructions()`, we handle such situations where if we see a loop body before the loop-top, we disregard such loop from getting aligned, but since `bbNatLoopNum` was not up to date, we were failing to recognize it. With the fix, we will unmark such loops from participating in alignment.

Fixes: https://github.com/dotnet/runtime/issues/82729

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
Didn't we recently make some change to SPMI to hide diffs from alignment changes? Seems like you might want to run asmdiffs locally with this disabled to be sure nothing unexpected is happening.

**Comment by kunalspathak** [comment]
yes, I forgot about it. pushed.

**Comment by kunalspathak** [comment]
> yes, I forgot about it. pushed.

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=209417&view=ms.vss-build-web.run-extensions-tab).

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -2389,6 +2389,15 @@ void Compiler::fgCompactBlocks(BasicBlock* block, BasicBlock* bNext)
...
 
+    if ((bNext->bbJumpDest != nullptr) && bNext->bbJumpDest->isLoopAlign())
+    {
+        // `bNext` has a backward target to some block which mean bNext is part of a loop.
+        // `block` into which `bNext` is compacted should be updated with its loop number
+        JITDUMP("Updating loop number for " FMT_BB " from " FMT_LP " to " FMT_LP ".\n", block->bbNum,
+                block->bbNatLoopNum, bNext->bbNatLoopNum);
+        block->bbNatLoopNum = bNext->bbNatLoopNum;
+    }
+
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgopt.cpp:2400]: Should we also update the loop table -- presumably this loop now has a new top? Or can this only happen when we no longer are maintaining the loop table?

Comment (kunalspathak) [reply src/coreclr/jit/fgopt.cpp:2400]: We do update loop table by calling `fgUpdateLoopsAfterCompacting()` below.

**Review by AndyAyersMS — APPROVED** [review]

**Pull request merged**